broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageDataStore.java

```diff
@@ -32,6 +32,18 @@ public void addMessage(MessageEntry message) {
     getData().messages.add(message);
   }
 
+  public boolean hasMessage(MessageEntry message) {
+    return getData()
+        .messages
+        .stream()
+        .anyMatch(
+            m ->
+                m.getId() != null
```

Comment:
If a message can't be null, shouldn't we check for empty string here then? My understanding of `BaseProperty.hasValue` in this case means all messages will always have a value, and therefore every `MessageEntry` will be created with either a normal ID, or an empty string, meaning two messages with empty string IDs, same name, same correlation key, would be the same?

Reply:
You're right. I fixed the check if the message id is set -> a099b72